synapse/api/auth/msc3861_delegated.py

```diff
@@ -282,17 +281,6 @@ async def get_user_by_req(
                             "Impersonation not possible by a non admin user",
                         )
 
-            # Deny the request if the user account is locked.
-            if not allow_locked and await self.store.get_user_locked_status(
-                requester.user.to_string()
-            ):
-                raise AuthError(
-                    401,
-                    "User account has been locked",
-                    errcode=Codes.USER_LOCKED,
-                    additional_fields={"soft_logout": True},
-                )
-
```

Comment:
Note that the same logic remains in auth/internal.py

Reply:
Before this was removed, I think `self.store.get_user_locked_status` would raise a 404 error because the dummy user doesn't exist?